provider/sourcegraph-search/index.ts

```diff
@@ -0,0 +1,131 @@
+import type {
+    AnnotationsParams,
+    AnnotationsResult,
+    CapabilitiesParams,
+    CapabilitiesResult,
+    Item,
+    ItemsParams,
+    ItemsResult,
+    Provider,
+    ProviderSettings,
+} from '@openctx/provider'
+import { SourcegraphGraphQLAPIClient } from './graphql'
+import { isError, searchForFileChunks } from './search'
+
+const graphqlClient = new SourcegraphGraphQLAPIClient()
+
+interface ItemsParamsProposed extends ItemsParams {
+    candidate?: Item
+}
+
+interface ProviderProposed<S extends {} = ProviderSettings> extends Provider {
+    candidateItems?(params: ItemsParamsProposed, settings: S): ItemsResult | Promise<ItemsResult>
+    items?(params: ItemsParamsProposed, settings: S): ItemsResult | Promise<ItemsResult>
+}
+
+/** Used to prefix context queries to indicate a sourcegraph search */
+const QUERY_PREFIXES = ['?', 'src:']
+
+/** We always limit our searches to files and a conservatively sized result set */
+const SEARCH_QUERY_CONST = 'type:file count:10'
+
+const sourcegraphSearch: ProviderProposed = {
+    capabilities(params: CapabilitiesParams, settings: ProviderSettings): CapabilitiesResult {
+        return {
+            // empty since we don't provide any annotations.
+            selector: [],
+        }
+    },
```

Comment:
FYI I haven't added item kind yet, since not exactly sure on the shape of the API. But should be easy to add once it is there.